plutus-core/plutus-ir/test/transform/inline/letFunConstMulti.golden

```diff
@@ -5,5 +5,5 @@
     (vardecl constFun (fun (con integer) (fun (con integer) (con integer))))
     (lam x (con integer) (lam y (con integer) x))
   )
-  [ constFun (con integer 3) ]
+  [ constFun [ (lam y (con integer) (con integer 3)) (con integer 5) ] ]
```

Comment:
right, looking into it.

Reply:
This is actually the right before. Terms are processed bottom-up, so `constFun 3` is processed first. It has a bigger size then `\y -> 3`, so `constFun` is inlined here, yielding `(\y -> 3) 5`. The `5` will be eliminated by dead binding elimination in the next round.

Reply:
> Okay, that does make sense, but I would also expect the other constFun to be inlined, no?

No, because of the types.
```
size (constFun 3) = 3
size (\(y :: Integer) -> 3)) = 3

size (constFun ((\(y :: Integer) -> 3) 5)) = 7
size (\(y' :: Integer -> Integer) -> ((\(y :: Integer) -> 3) 5)) = 9
```

The `Integer -> Integer` adds 3 to the size.